x/merkledb/trie.go

```diff
@@ -13,7 +13,21 @@ import (
 
 var errNoNewRoot = errors.New("there was no updated root in change list")
 
+type MerkleRootGetter interface {
+	// GetMerkleRoot returns the merkle root of the Trie
+	GetMerkleRoot(ctx context.Context) (ids.ID, error)
+}
+
+type ProofGetter interface {
```

Comment:
Made these interfaces so we don't need to declare the same method in `ReadOnlyTrie` and `SyncableDB`